editor/src/clj/editor/code/script_compilation.clj

```diff
@@ -147,31 +147,48 @@
   (loop [cursor-ranges (transient [])
          tokens (lua-parser/tokens (data/lines-reader lines))
          paren-count 0
-         consumed []]
+         consumed []
+         skipped nil]
     (if-some [[text :as token] (first tokens)]
       (case (count consumed)
-        0 (recur cursor-ranges (next tokens) 0 (case text "go" (conj consumed token) []))
-        1 (recur cursor-ranges (next tokens) 0 (case text "." (conj consumed token) []))
-        2 (recur cursor-ranges (next tokens) 0 (case text "property" (conj consumed token) []))
+        0 (recur cursor-ranges (next tokens) 0 (case text "go" (conj consumed token) []) skipped)
+        1 (recur cursor-ranges (next tokens) 0 (case text "." (conj consumed token) []) skipped)
+        2 (recur cursor-ranges (next tokens) 0 (case text "property" (conj consumed token) []) skipped)
         3 (case text
-            "(" (recur cursor-ranges (next tokens) (inc paren-count) consumed)
+            "(" (recur cursor-ranges (next tokens) (inc paren-count) consumed skipped)
             ")" (let [paren-count (dec paren-count)]
                   (assert (not (neg? paren-count)))
                   (if (pos? paren-count)
-                    (recur cursor-ranges (next tokens) paren-count consumed)
+                    (recur cursor-ranges (next tokens) paren-count consumed skipped)
                     (let [next-tokens (next tokens)
                           [next-text :as next-token] (first next-tokens)
                           [_ start-row start-col] (first consumed)
                           [end-text end-row end-col] (if (= ";" next-text) next-token token)
                           end-col (+ ^long end-col (count end-text))
-                          start-cursor (data/->Cursor start-row start-col)
                           end-cursor (data/->Cursor end-row end-col)
-                          cursor-range (data/->CursorRange start-cursor end-cursor)]
-                      (recur (conj! cursor-ranges cursor-range)
-                             next-tokens
-                             0
-                             []))))
-            (recur cursor-ranges (next tokens) paren-count consumed)))
+                          start-cursor (data/->Cursor start-row start-col)]
+                      (if (nil? skipped)
+                        (let [cursor-range (data/->CursorRange start-cursor end-cursor)]
+                          (recur (conj! cursor-ranges cursor-range)
+                                 next-tokens
+                                 0
+                                 []
+                                 nil))
+                        (let [[_ end-row-skipped end-col-skipped] skipped
```

Comment:
If we have a token that we want to skip, two cursor ranges should be produced: one for the "before skip" token and another for `)` or `);`.